plugins/maven/src/main/java/org/jetbrains/idea/maven/importing/MavenModuleNameMapper.java

```diff
@@ -74,19 +78,19 @@ private static void resolveModuleNames(Collection<MavenProject> projects,
     for (NameItem nameItem : names) {
       if (nameItem.module == null) {
 
-        Integer c = nameCounters.get(nameItem.originalName);
+        Integer c = nameCountersLowerCase.get(nameItem.originalName.toLowerCase(ROOT));
 
         if (c != null) {
           nameItem.number = c;
-          nameCounters.put(nameItem.originalName, c + 1);
+          nameCountersLowerCase.put(nameItem.originalName.toLowerCase(ROOT), c + 1);
         }
 
         do {
           String name = nameItem.getResultName();
           if (existingNames.add(name)) break;
 
           nameItem.number++;
-          nameCounters.put(nameItem.originalName, nameItem.number + 1);
+          nameCountersLowerCase.put(nameItem.originalName.toLowerCase(ROOT), nameItem.number + 1);
```

Comment:
it could make sense to store new field `originalNameLowerCase` in the class `NameItem`